### DISCUSSION TITLE
WPF App is usually hanging

### DISCUSSION THREAD

**Discussion by phuocminh6464** [cefsharp/CefSharp#4812] [General]
Hi everyone, I have an WPF with .NET 7 and using CefSharp.Wpf.NETCore version 120.2.70. There are many buttons in the MainWindow screen and each button will show a different web url with the header on top. The app is hanging in 2 cases:

**Case 1:**
When users click to any button in MainWindow then go to web page, they are able to return MainWindow by clicking home button in the header. At that time, I use the following code to release the current web page and move to MainWindow:
```
 browser.GetBrowserHost().CloseBrowser(true);
 browser.Dispose();
 MainWindow window = new MainWindow();
 window.Show();
 this.Hide();
```
But the app sometimes is hanging when users click button on MainWindow after loaded web screen so users must close app to start again.

**Case 2:** 
There is 1 button go to a website which allows users input data and using ```CefSharp.PostMessage();``` to send message to WPF for printing. After printed, the WPF app must return the print result to web so I create 2 window for this flow.

The 1st window is a webpage send data to WPF and then opening the 2nd window which will overlay the 1st. The 2nd window will print and return result to web then closing immediately. The app is ALWAYS hanging in the printing step. I don't know what exactly makes app is hanging because there is 1 screen allows to test print with the fixed data and it's not hanging here.

I'm using the below code to send result to web:
```
string script = @" var event= new CustomEvent('messageReceiveEvent', {detail: " + jsonObj + @"});
                   document.dispatchEvent(event);"; 
browser.GetMainFrame().ExecuteJavaScriptAsync(script);
```

Anyone can advise me the solution to resole hanging issuse here? Thanks.

**Comment by phuocminh6464** [comment]
Can anyone help me?

**Comment by amaitland** [comment]
> CefSharp.Wpf.NETCore version 120.2.70

Does the problem reproduce in the current supported version? At time of writing M124

Check the log file to see if there's any errors. https://github.com/cefsharp/CefSharp/wiki/Trouble-Shooting#log-file

Without an actual example it's impossible to say what's going on. 

You can fork https://github.com/cefsharp/CefSharp.MinimalExample to use as a base for examples.

**Reply by phuocminh6464** [reply]
> Does the problem reproduce in the current supported version? At time of writing M124

Sorry, I haven't check with version 124 yet. I installed the same app in many machines but only 2 machines is hanging, the others is good so i think the version is not root reason.

> Check the log file to see if there's any errors. https://github.com/cefsharp/CefSharp/wiki/Trouble-Shooting#log-file
> Without an actual example it's impossible to say what's going on.

Thank you, I will check the log file

**Update status:**
In the scenario 2, I create a new thread for printing (the 2nd window) and app is not hanging anymore. When I return to MainWindow and go to web page again, every thing is okay. But hanging occurs again if app send an event to webpage.

**Reply by phuocminh6464** [reply]
I'm not know the reason but after upgrade RAM, the issue is solved. Thank @amaitland so much for supporting
